app/uk/gov/hmrc/agentsubscriptionfrontend/controllers/CheckAgencyController.scala

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2017 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.agentsubscriptionfrontend.controllers
+
+import javax.inject.{Inject, Singleton}
+
+import play.api.data.Forms.{mapping, _}
+import play.api.data.validation._
+import play.api.data.{Form, Mapping}
+import play.api.i18n.{I18nSupport, MessagesApi}
+import play.api.mvc.{AnyContent, Request, _}
+import uk.gov.hmrc.agentsubscriptionfrontend.auth.{AuthActions, NoOpRegime}
+import uk.gov.hmrc.agentsubscriptionfrontend.config.AppConfig
+import uk.gov.hmrc.agentsubscriptionfrontend.connectors.AgentSubscriptionConnector
+import uk.gov.hmrc.agentsubscriptionfrontend.models.Registration
+import uk.gov.hmrc.agentsubscriptionfrontend.views.html
+import uk.gov.hmrc.play.frontend.auth.AuthContext
+import uk.gov.hmrc.play.frontend.auth.connectors.AuthConnector
+import uk.gov.hmrc.play.frontend.controller.FrontendController
+
+import scala.concurrent.Future
+
+@Singleton
+class CheckAgencyController @Inject()
+  (override val messagesApi: MessagesApi, override val authConnector: AuthConnector, val agentSubscriptionConnector: AgentSubscriptionConnector)
+  (implicit appConfig: AppConfig)
+  extends FrontendController with I18nSupport with AuthActions {
+
+  private val knownFactsForm = Form[KnownFacts](
+    mapping(
+      "utr" -> FieldMappings.utr,
+      "postcode" -> FieldMappings.postcode
+    )(KnownFacts.apply)(KnownFacts.unapply)
+  )
+
+  private[controllers] val showCheckAgencyStatusBody: AuthContext => (Request[AnyContent]) => Future[Result] = {
+    implicit authContext =>
+      implicit request =>
+          Future successful Ok(html.check_agency_status(knownFactsForm))
+  }
+
+  val showAlreadySubscribed: Action[AnyContent] = AuthorisedWithSubscribingAgent { implicit authContext => implicit request =>
```

Comment:
Should just be this one and the hasOtherEnrolments endpoint. I haven't written the tests yet, as they will come with the meat of this story.